### PR TITLE
fix: skip generating error events for unauthorized or forbidden errors

### PR DESCRIPTION
**Pull request by duyhungtnn**
part of https://github.com/bucketeer-io/bucketeer/issues/1284

### Demo
Screenshot 2024-11-22 at 14 33 58

### CODE REVIEW

File: src/internal/event/EventInteractor.ts [hunk 1]
```diff
@@ -169,6 +169,17 @@ export class EventInteractor {
   }
 
   trackFailure(apiId: ApiId, featureTag: string, error: BKTException): void {
+
+    if (error.type === MetricsEventType.UnauthorizedError) {
+      console.log('An unauthorized error occurred. Please check your API Key.')
```
Comment (cre8ivejp) [inline src/internal/event/EventInteractor.ts:174]: How about using the error level?

```suggestion
      console.error('An unauthorized error occurred. Please check your API Key.')
```

File: src/internal/event/EventInteractor.ts [hunk 2]
```diff
@@ -169,6 +169,17 @@ export class EventInteractor {
...
+      return
+    }
+
+    if (error.type === MetricsEventType.ForbiddenError) {
+      console.log('An forbidden error occurred. Please check your API Key.')
```
Comment (cre8ivejp) [inline src/internal/event/EventInteractor.ts:179]: Same here.

```suggestion
      console.log('An forbidden error occurred. Please check your API Key.')
```

**Comment by duyhungtnn** [comment]
@cre8ivejp I updated to use console.error()

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**